src/main/java/io/stargate/sgv2/jsonapi/api/v1/metrics/JsonShreddingMetricsReporter.java

```diff
@@ -0,0 +1,37 @@
+package io.stargate.sgv2.jsonapi.api.v1.metrics;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+import io.stargate.sgv2.jsonapi.service.shredding.model.WritableShreddedDocument;
+import jakarta.inject.Inject;
+
+/** Reports the time and size of json shredding. */
+public class JsonShreddingMetricsReporter {
+
+  private final MeterRegistry meterRegistry;
+  private final JsonApiMetricsConfig jsonApiMetricsConfig;
+
+  private Timer.Sample sample;
+
+  @Inject
```

Comment:
Remove this `@Inject`: instances are created by the factory, not injected by framework.